src/Bridge/Doctrine/Orm/Filter/SearchFilter.php

```diff
@@ -139,27 +138,26 @@ protected function filterProperty(string $property, $value, QueryBuilder $queryB
              * The only fix is to provide field type to doctrine ...
              * BUT it is not possible with setParameter(), as third arg can only be
              * \Doctrine\DBAL\Connection::PARAM_STR_ARRAY
-             *
              * The only way to do this will be to rewrite the IN() statement to multiple values,
              * each provided as a single setParameter()
              */
 
             // let's do a very simple query if we don't need the complex one (99% of time)
-            $type = $metadata->getTypeOfField($field);
-            if (\in_array($type, ['string', 'integer'], true)) {
+            if (\in_array($metadata->getTypeOfField($field), ['string', 'integer'], true)) {
+                $valueParameter = $queryNameGenerator->generateParameterName($field);
                 $queryBuilder
                     ->andWhere(sprintf($wrapCase('%s.%s').' IN (:%s)', $alias, $field, $valueParameter))
                     ->setParameter($valueParameter, $caseSensitive ? $values : array_map('strtolower', $values));
             } else {
                 $inQuery = [];
-                $nbArgs = \count($values);
-
-                // QueryBuilder->setParameters() reset current parameters set, so we need to merge them all together
-                $parameters = $queryBuilder->getParameters();
-
-                for ($i = 0; $i < $nbArgs; ++$i) {
-                    $inQuery[] = ':filterproperty'.$i;
-                    $parameters->add(new Parameter('filterproperty'.$i, $caseSensitive ? $values[$i] : strtolower($values[$i]), $type));
+                $args = [];
```

Comment:
Why not always using this conditional branch instead of having an if / else?
It will:
- make the code easier to read (and maintain),
- prevent bugs to arise only in one conditional branch.
WDYT @soyuka @dunglas?

Reply:
This will require updating some tests expectations because when multiple values are given, they are set as one unique array parameter, and that will become several parameters. See https://github.com/api-platform/core/pull/3774/checks?check_run_id=1379433208#step:14:91

I'm currently blocked at adding UUID tests, I cant fully understand yet how tests are organized and how can I add custom types.
```
\Doctrine\DBAL\Types\Type::addType('uuid', \Ramsey\Uuid\Doctrine\UuidType::class);
```
is required to manipulate uuid types.